### PR TITLE
web api

### PR DESCRIPTION
**Pull request by mike-allaway**
Purpose of this PR is to develop the bulk of the web API.

Note that the web API won't be complete in this PR, as it depends on other, future work (e.g. development of custom user accounts).

Also makes minor changes to other parts of Django to get the API working (e.g. the settings files, requirements, etc.) and have tagged on the accessibility change (from an embedded page to an external page) as only minor.

### CODE REVIEW

File: django/journal/api.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from . import models, serializers
+from rest_framework import viewsets, permissions
+
+
+class ProjectViewSet(viewsets.ModelViewSet):
+    """
+    Creates a ViewSet for managing CRUD operations on the Project model
+    """
+
+    queryset = models.Project.objects.all()
```
Comment (branfosj) [inline django/journal/api.py]: The models all have a `admin_published` field. Should the api be restricting to showing only when this is true?

**Review by branfosj — DISMISSED** [review]
One query visibility of items in the API. Otherwise I am happy with this.

File: django/journal/api.py (hunk 1 above)
Comment (mike-allaway) [reply django/journal/api.py]: Good spot, thanks. Have fixed this and tested it's working as expected.

**Review by branfosj — APPROVED** [review]
👍

**Pull request merged**